pkg/cloud/api/path.go

```diff
@@ -82,6 +125,13 @@ func (p Path) HasPrefix(prefix Path) bool {
 
 	var i int
 	for i = range prefix {
+		// anyIndex matches any index.
```

Comment:
Use isMatch() as mentioned above.

Reply:
is match is bidirectional if we have 2 paths one or the other might be "anyIndex". In HasPrefix only prefix can be anyIndex and we expect `p` to be slice/map index with actual value.
I will add check for values to `isSliceIndex` and `isMapIndex` funcions.
WDYT?

Reply:
done